spec/11-statements.md

```diff
@@ -903,6 +903,10 @@ by value or byRef. If *expression* is omitted the value `NULL` is used.
 If execution flows into the closing brace (`}`) of a function, `return
 NULL;` is implied.
 
+Explicit `return` statements with *expression* given are not permitted within a
+function with a `void` [return type](13-functions.md#return-typing) and MUST
+cause a compile-time fatal error.
```

Comment:
I think this is the only part of the spec that mentions "compile-time fatal error" aside from the definition of `E_COMPILE_ERROR`. Is this the best way to express it? I'm trying to make it clear that this error doesn't occur when `return` is executed, but rather when the statement is encountered when parsing or compiling.